### PR TITLE
Fix handling of active session count for Spanner session pool

### PR DESCRIPTION
**Pull request by jskeet**
I've tried to add enough comments to explain this - let me know if any don't quite work for you (thinking long-term).

### CODE REVIEW

File: apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/PoolRewrite/SessionPool.TargetedSessionPool.cs [hunk 1]
```diff
@@ -90,24 +90,27 @@ internal TargetedSessionPool(SessionPool parent, DatabaseName databaseName, bool
...
-                // FIXME: We don't cope with the "Minimum pool size of 0, max active = 100, currently active = 99" case.
-                StartAcquisitionTasksIfNecessary();
-
                 bool success = false;
                 try
                 {
-                    Interlocked.Increment(ref _activeSessionCount);
-                    PooledSession session = await AcquireSessionImplAsync(transactionOptions, cancellationToken).ConfigureAwait(false);
+                    int sessionCount = Interlocked.Increment(ref _activeSessionCount);
+                    if (sessionCount > Options.MaximumActiveSessions && Options.WaitOnResourcesExhausted == ResourcesExhaustedBehavior.Fail)
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/PoolRewrite/SessionPool.TargetedSessionPool.cs]: This is the previous active session count, so I think this needs to be `>=`

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/SessionPoolTests.TargetedSessionPoolTests.cs [hunk 2]
```diff
@@ -197,6 +198,71 @@ public async Task ReleaseSession_SessionEvicted()
...
+            public async Task AcquireAsync_ZeroPoolSizeAcquireUpToMaximum()
+            {
+                var pool = CreatePool(false);
+                var client = (SessionTestingSpannerClient)pool.Client;
+                pool.Options.MinimumPooledSessions = 0;
+
+                await client.Scheduler.RunAsync(async () =>
+                {
+                    await AcquireAllSessionsAsync(pool);
+                });
```
Comment (amanda-tarafa) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/SessionPoolTests.TargetedSessionPoolTests.cs]: `Assert.Equal(pool.Options.MaximumActiveSessions, pool.ActiveSessionCount);`?

Comment (amanda-tarafa) [inline apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/SessionPoolTests.TargetedSessionPoolTests.cs]: `Assert.Equal(pool.Options.MaximumActiveSessions, pool.ActiveSessionCount);`?

File: apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/PoolRewrite/SessionPool.TargetedSessionPool.cs (hunk 1 above)
Comment (jskeet) [reply apis/Google.Cloud.Spanner.V1/Google.Cloud.Spanner.V1/PoolRewrite/SessionPool.TargetedSessionPool.cs]: Nope, Interlocked.Increment returns the new count. Do you think it's worth a reminder of that as a comment?

File: apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/SessionPoolTests.TargetedSessionPoolTests.cs (hunk 2 above)
Comment (jskeet) [reply apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.Tests/V1/SessionPoolTests.TargetedSessionPoolTests.cs]: Yup, can do.

**Pull request merged**